pkg/covermerger/covermerger.go

```diff
@@ -182,32 +173,89 @@ func MergeCSVData(config *Config, reader io.Reader) (map[string]*MergeResult, er
 	return mergeResult, nil
 }
 
-func mergeChanData(c *Config, recordsChan <-chan FileRecord) (map[string]*MergeResult, error) {
+type FileRecords struct {
+	fileName string
+	records  []FileRecord
+}
+
+func mergeChanData(c *Config, recordChan <-chan FileRecord) (map[string]*MergeResult, error) {
+	processedFiles := map[string]int{}
+	frecordChan := make(chan FileRecords)
+	exitCh := make(chan struct{})
+	go func() {
```

Comment:
Nit: I'd consider splitting this goroutine code into another function. It's somewhat independent + `mergeChanData()` is already too big.

Reply:
done